src/entities/activity/ui/ActivityProgressPreviewList.tsx

```diff
@@ -0,0 +1,28 @@
+import { ActivityListItem } from "../lib"
+import { getRandomInt } from "../lib/getRandomInt"
+import { ActivityProgressPreviewCard } from "./ActivityProgressPreviewCard"
+
+interface ActivityProgressPreviewListProps {
+  activities: ActivityListItem[]
+}
+
+export const ActivityProgressPreviewList = ({ activities }: ActivityProgressPreviewListProps) => {
+  const activitiesPreview = activities.map(activity => {
+    const itemsLength = 10 // activity.items.length in the real implementation
```

Comment:
Also, please move the mapper function to the `model` level

Reply:
Done